gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GCSManualClientImpl.java

```diff
@@ -0,0 +1,222 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.auth.Credentials;
+import com.google.cloud.hadoop.util.AccessBoundary;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.flogger.GoogleLogger;
+import java.io.IOException;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Provides read/write access to Google Cloud Storage (GCS), using Java nio channel semantics. This
+ * is a basic implementation of the GoogleCloudStorage interface that mostly delegates through to
+ * the appropriate API call(s) google-cloud-storage client.
+ */
+public class GCSManualClientImpl implements GoogleCloudStorage {
```

Comment:
"manual" client do not sound right. Should we call this GoogleCloudStorageImplV2?

Reply:
I don't prefer "V2" as it's vague, I can go with 'java client' as it's manual java client and that way it will be consistent with our flag name too.